packages/validator/src/services/doppelgangerService.ts

```diff
@@ -103,7 +100,7 @@ export class DoppelgangerService {
     const indicesToCheckMap = new Map<ValidatorIndex, PubkeyHex>();
 
     for (const [pubkeyHex, state] of this.doppelgangerStateByPubkey.entries()) {
-      if (state.remainingEpochs > 0) {
+      if (state.remainingEpochs > 0 && state.nextEpochToCheck <= currentEpoch) {
```

Comment:
Is it not redundant condition? Is not`remainingEpochs` already related to `currentEpoch`?

Reply:
There is a difference, if a validator is registered first time `state.remainingEpochs` will always be > 0 (unless pre-genesis)
https://github.com/ChainSafe/lodestar/blob/13978344a312bd5238e11f26839016331a382799/packages/validator/src/services/doppelgangerService.ts#L64

but `state.nextEpochToCheck` will always be current epoch + 1.
https://github.com/ChainSafe/lodestar/blob/13978344a312bd5238e11f26839016331a382799/packages/validator/src/services/doppelgangerService.ts#L65

This means that the first time we check liveness for a validator it is for a epoch that is not relevant because it will be skipped anyways (see PR description).

The main goal of checking the epoch here is to skip earlier, this avoids unnecessary liveness requests to the beacon node.